bundles/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/InfluxDBPersistenceService.java

```diff
@@ -0,0 +1,264 @@
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is the implementation of the InfluxDB {@link PersistenceService}. It persists item values
+ * using the <a href="http://influxdb.org">InfluxDB time series database. The states (
+ * {@link State}) of an {@link Item} are persisted by default in a time series with names equal to the name of
+ * the item.
+ *
+ * This addon supports 1.X and 2.X versions, as two versions are incompatible and use different drivers the
+ * specific code for each version is accessed by {@link InfluxDBRepository} and {@link FilterCriteriaQueryCreator}
+ * interfaces and specific implementation reside in {@link org.openhab.persistence.influxdb.internal.influx1} and
+ * {@link org.openhab.persistence.influxdb.internal.influx2} packages
+ *
+ * @author Theo Weiss - Initial contribution, rewrite of org.openhab.persistence.influxdb
+ * @author Joan Pujol Espinar - Addon rewrite refactoring code and adding support for InfluxDB 2.0. Some tag code is
+ *         based
+ *         from not integrated branch from Dominik Vorreiter
+ */
+@NonNullByDefault
+@Component(service = { PersistenceService.class,
+        QueryablePersistenceService.class }, configurationPid = "org.openhab.influxdb", property = {
+                Constants.SERVICE_PID + "=org.openhab.influxdb",
+                ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=persistence:influxdb",
+                ConfigurableService.SERVICE_PROPERTY_LABEL + "=InfluxDB persistence layer",
+                ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=persistence" })
+public class InfluxDBPersistenceService implements QueryablePersistenceService {
+    public static final String SERVICE_NAME = "influxdb";
+
+    private final Logger logger = LoggerFactory.getLogger(InfluxDBPersistenceService.class);
+
+    // External dependencies
+    @Nullable
+    private ItemRegistry itemRegistry;
+    @Nullable
+    private MetadataRegistry metadataRegistry;
+
+    // Internal dependencies/state
+    private InfluxDBConfiguration configuration = InfluxDBConfiguration.NO_CONFIGURATION;
+    @NonNullByDefault({}) // Relax rules because can only be null if component is not active
+    private ItemToStorePointCreator itemToStorePointCreator;
+    @NonNullByDefault({}) // Relax rules because can only be null if component is not active
+    private InfluxDBRepository influxDBRepository;
+
+    @Activate
+    /**
+     * Connect to database when service is activated
+     */
+    public void activate(final @Nullable Map<String, @Nullable Object> config) {
```

Comment:
Looked like the annotation was flying away there.
```suggestion
    /**
     * Connect to database when service is activated
     */
     @Activate
    public void activate(final @Nullable Map<String, @Nullable Object> config) {
```